pkg/services/alerting/engine.go

```diff
@@ -115,7 +121,13 @@ func (e *Engine) processJob(grafanaCtx context.Context, job *Job) error {
 
```

Comment:
if there's an error (panic) during execution, you should also mark the span as failed, which you can do with
```
ext.Error.Set(span, true) 
```
and then also log the error into the span with:
```
span.LogFields(log.Error(err))
or something like:
func Errorf(span opentracing.Span, format string, a ...interface{}) {
        span.LogFields(log.Error(fmt.Errorf(format, a...)))
}
```
this is super useful anywhere there's an error, as they will show up as errors in the tracing UI as well.

BTW probably not related to this pr, but why twice the same recover defer within processJob ?

Reply:
the first recover wont catch panic's from the sub functions. 

```go
go func() {
  panic("help!")
}()
```
note that we start the go routine straight away